Add meta-documentation about Gatsby (#878)
grafana/k6-docs: README.md

# k6 Documentation

![Staging](https://github.com/grafana/k6-docs/workflows/Staging/badge.svg)
![Production](https://github.com/grafana/k6-docs/workflows/Production/badge.svg)

Welcome to the k6 docs!
This repo contains the source code of the [k6 documentation](https://k6.io/docs/).

Some key facts:

> - 💙 Issues and contributions are always welcome!

Author: ```suggestion
- Issues and contributions are **always welcome**!
```

If you find something that you think you can fix, please go ahead. You don't need to ask permission.

Markdown files for the documentation are located in [`src/data/markdown/docs`](src/data/markdown/docs) folder. The URL structure is generated based on the folder structure and file names.

For small changes and spelling fixes, the GitHub UI is most convenient. 
For larger contributions, consider running the project locally to see how the changes look like before making a pull request.
You can build locally with Docker, as described in the following section,
or you can use node, as described in the [Contributor's Guide](./contributors-guide)
 
The following docs cover the different stages of the writing workflow:

- [Build locally](./contributors-guide#build-locally). For any large changes, being able to view how the site looks live hugely improves writer comfort and doc quality. To build locally, you'll need node and a node version manager.
- [Contributing reference](./contributors-guide/gatsby-reference.md). The syntax and structure of the page paths, custom style components, program redirects, and so on.

> - [Troubleshooting](./contributors-guide/troubleshooting.md). Unfortunately, things can go wrong, and Gatsby can be tough on errors. Fortunately, the inscrutable Gatsby errors usually share the same few fixes.

Author: ```suggestion
- [Troubleshooting](./contributors-guide/troubleshooting.md). Unfortunately, things can go wrong, and Gatsby can be tough on errors. Fortunately, most errors can be solved by the same few fixes.
```
